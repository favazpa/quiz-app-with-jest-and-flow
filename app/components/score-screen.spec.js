// @flow
import React from 'react';
import {Component} from 'react';
import ScoreScreen from './score-screen';
import ShallowRenderer from 'react-test-renderer/shallow';

let renderer = new ShallowRenderer();

describe('src/component/scoreScreen', () => {
  const renderButton = (score) => {
    renderer.render(<ScoreScreen score={score} />);
    return renderer.getRenderOutput();
  };

  test('should render correctly', () => {
    const result = renderButton(4);
    expect(result).toMatchSnapshot();
  });

  test('should render correctly', () => {
    const result = renderButton(1);
    expect(result).toMatchSnapshot();
  });
 
});