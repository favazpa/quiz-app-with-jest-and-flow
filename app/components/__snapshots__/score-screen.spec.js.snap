// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/scoreScreen should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1f2336",
      "height": "100%",
      "justifyContent": "center",
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#d8e5eb",
        "borderRadius": 25,
        "height": "70%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source=""
      style={
        {
          "height": "30%",
          "margin": 20,
          "width": "70%",
        }
      }
    />
    <Text
      style={
        {
          "color": "black",
          "fontSize": 28,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Congrats!
    </Text>
    <Text
      style={
        [
          {
            "color": "#05b052",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 20,
          },
          {
            "color": "#05b052",
          },
        ]
      }
    >
      80
      % Score
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Quiz completed successfully
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "textAlign": "center",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        You attempt 
      </Text>
      <Text
        style={
          [
            {
              "color": "black",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
            {
              "color": "#2257a3",
              "fontWeight": "800",
            },
          ]
        }
      >
        5 questions
         
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        and from that 
      </Text>
      <Text
        style={
          [
            {
              "color": "black",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
            {
              "color": "#05b052",
              "fontWeight": "800",
            },
          ]
        }
      >
        4
         answer
         
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        is correct
      </Text>
    </Text>
  </View>
  <Button
    style={
      {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 15,
        "bottom": 20,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 20,
        "position": "absolute",
        "width": "90%",
      }
    }
    tilteStyle={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "900",
      }
    }
    title="Retry"
  />
</View>
`;

exports[`src/component/scoreScreen should render correctly 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1f2336",
      "height": "100%",
      "justifyContent": "center",
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#d8e5eb",
        "borderRadius": 25,
        "height": "70%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="contain"
      source=""
      style={
        {
          "height": "30%",
          "margin": 20,
          "width": "70%",
        }
      }
    />
    <Text
      style={
        {
          "color": "black",
          "fontSize": 28,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Better luck next time!
    </Text>
    <Text
      style={
        [
          {
            "color": "#05b052",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 20,
          },
          {
            "color": "red",
          },
        ]
      }
    >
      20
      % Score
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Quiz completed successfully
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "textAlign": "center",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        You attempt 
      </Text>
      <Text
        style={
          [
            {
              "color": "black",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
            {
              "color": "#2257a3",
              "fontWeight": "800",
            },
          ]
        }
      >
        5 questions
         
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        and from that 
      </Text>
      <Text
        style={
          [
            {
              "color": "black",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
            {
              "color": "#05b052",
              "fontWeight": "800",
            },
          ]
        }
      >
        1
         answer
         
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        is correct
      </Text>
    </Text>
  </View>
  <Button
    style={
      {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 15,
        "bottom": 20,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 20,
        "position": "absolute",
        "width": "90%",
      }
    }
    tilteStyle={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "900",
      }
    }
    title="Retry"
  />
</View>
`;
